### ISSUE TITLE
compute/v2/flavors: error parsing empty swap

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#7]
A flavor being returned by OpenStack looks like:

```json
    {
      "OS-FLV-DISABLED:disabled": false,
      "OS-FLV-EXT-DATA:ephemeral": 0,
      "disk": 160,
      "id": "5",
      "links": [
        {
          "href": "http://10.1.11.212:8774/v2.1/80839b1b20aa41a3a4bf32718bae67cb/flavors/5",
          "rel": "self"
        },
        {
          "href": "http://10.1.11.212:8774/80839b1b20aa41a3a4bf32718bae67cb/flavors/5",
          "rel": "bookmark"
        }
      ],
      "name": "m1.xlarge",
      "os-flavor-access:is_public": true,
      "ram": 16384,
      "rxtx_factor": 1,
      "swap": "",
      "vcpus": 8
    },
```

When no swap is set, the returned value is `""` which triggers the error:

```
Unable to retrieve flavors: json: cannot unmarshal string into Go value of type int
```

`Swap` is defined in Gophercloud as an [int](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/flavors/results.go#L44), as it should, because Nova defines it as an [int](https://github.com/openstack/nova/blob/e8cdef91f82f3a46d6f195d96cbd450934119022/nova/compute/flavors.py#L52-L63) as well. 

Was [mapstructure](https://github.com/rackspace/gophercloud/blob/master/openstack/compute/v2/flavors/results.go) doing more lenient casting of values?

**Comment by jrperritt** [comment]
Just another misguided OpenStack design...

>> Was mapstructure doing more lenient casting of values?

Optionally, yes. There was a `WeakDecode` method that would handle some conversions like `int` -> `string` for you.

I guess the way to do this is to define an `UnmarshalJSON` method on `Flavor` (so that it implements `json.Unmarshaler` and handle (probably two-step it) it there.

Hopefully this won't be a recurring theme.

**Comment by jrperritt** [comment]
Closed by #12
